lib/color.js

```diff
@@ -219,6 +218,75 @@ class ColorXY {
         return kelvinToMireds(kelvin);
     }
 
+    /**
+     * Converts CIE color space to RGB color space
+     * From: https://github.com/usolved/cie-rgb-converter/blob/master/cie_rgb_converter.js
+     * @return {ColorRGB}
+     */
+    toRGB() {
+        // use maximum brightness
+        const brightness = 254;
+
+        const z = 1.0 - this.x - this.y;
+        const Y = (brightness / 254).toFixed(2);
+        const X = (Y / this.y) * this.x;
+        const Z = (Y / this.y) * z;
+
+        // Convert to RGB using Wide RGB D65 conversion
+        let red = X * 1.656492 - Y * 0.354851 - Z * 0.255038;
+        let green = -X * 0.707196 + Y * 1.655397 + Z * 0.036152;
+        let blue = X * 0.051713 - Y * 0.121364 + Z * 1.011530;
+
+        // If red, green or blue is larger than 1.0 set it back to the maximum of 1.0
+        if (red > blue && red > green && red > 1.0) {
+            green = green / red;
+            blue = blue / red;
+            red = 1.0;
+        } else if (green > blue && green > red && green > 1.0) {
+            red = red / green;
+            blue = blue / green;
+            green = 1.0;
+        } else if (blue > red && blue > green && blue > 1.0) {
+            red = red / blue;
+            green = green / blue;
+            blue = 1.0;
+        }
+
+        // Reverse gamma correction
+        // XXX: results in failed tests (bad tests?)
+        // red = red <= 0.0031308 ? 12.92 * red : (1.0 + 0.055) * Math.pow(red, (1.0 / 2.4)) - 0.055;
+        // green = green <= 0.0031308 ? 12.92 * green : (1.0 + 0.055) * Math.pow(green, (1.0 / 2.4)) - 0.055;
+        // blue = blue <= 0.0031308 ? 12.92 * blue : (1.0 + 0.055) * Math.pow(blue, (1.0 / 2.4)) - 0.055;
+
+
+        // Convert normalized decimal to decimal
+        red = Math.round(red * 255);
+        green = Math.round(green * 255);
+        blue = Math.round(blue * 255);
+
+        if (isNaN(red) || Object.is(red, -0)) {
+            red = 0;
+        }
+
+        if (isNaN(green) || Object.is(green, -0)) {
+            green = 0;
+        }
+
+        if (isNaN(blue) || Object.is(blue, -0)) {
+            blue = 0;
+        }
+
+        return new ColorRGB(red, green, blue);
```

Comment:
`ColorRGB` class uses normalized values (0..1), with (0..255) `.toXY()` and `.toHSV()` methods won't work. Also I'd omit any rounding in processing to avoid computation errors.

Reply:
Dropping the normalization to decimal probably will also get rid of the -0 I'm seeing 🤞

Reply:
Will push this change once I've done the other change requests, but... it's not looking good.

```
 FAIL  test/color.test.js
  ● lib/color.js › ColorXY › .toRGB() - red

    expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 3
    + Received  + 3

      Object {
    -   "blue": 0,
    -   "green": 0,
    -   "red": 1,
    +   "blue": 0.004251014680729584,
    +   "green": -0.005923094817025792,
    +   "red": 0.9999999999999999,
      }

      107 |             ['blue', {x: 0.1355, y: 0.0399}, {red: 0, green: 0, blue: 1.0}],
      108 |         ])('.toRGB() - %s', (_name, xy, rgb) => {
    > 109 |             expect(libColor.ColorXY.fromObject(xy).toRGB().toObject()).toStrictEqual(rgb);
          |                                                                        ^
      110 |         });
      111 |
      112 |         test.each([

      at test/color.test.js:109:72

  ● lib/color.js › ColorXY › .toRGB() - green

    expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 3
    + Received  + 3

      Object {
    -   "blue": 0,
    -   "green": 1,
    -   "red": 0,
    +   "blue": 0.00014235404949858123,
    +   "green": 0.9999999999999999,
    +   "red": -0.00035358849383455696,
      }

      107 |             ['blue', {x: 0.1355, y: 0.0399}, {red: 0, green: 0, blue: 1.0}],
      108 |         ])('.toRGB() - %s', (_name, xy, rgb) => {
    > 109 |             expect(libColor.ColorXY.fromObject(xy).toRGB().toObject()).toStrictEqual(rgb);
          |                                                                        ^
      110 |         });
      111 |
      112 |         test.each([

      at test/color.test.js:109:72

  ● lib/color.js › ColorXY › .toRGB() - blue

    expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 3
    + Received  + 3

      Object {
    -   "blue": 1,
    -   "green": 0,
    -   "red": 0,
    +   "blue": 0.9999999999999999,
    +   "green": 0.0005596045277429061,
    +   "red": -0.00012705215838029117,
      }

      107 |             ['blue', {x: 0.1355, y: 0.0399}, {red: 0, green: 0, blue: 1.0}],
      108 |         ])('.toRGB() - %s', (_name, xy, rgb) => {
    > 109 |             expect(libColor.ColorXY.fromObject(xy).toRGB().toObject()).toStrictEqual(rgb);
          |                                                                        ^
      110 |         });
      111 |
      112 |         test.each([

      at test/color.test.js:109:72
```

Reply:
After removing reverse gamma correction it seem about right, expect of slight calculation errors.

For values below 0 I'd suggest clipping it back at 0:

```diff
diff --git a/lib/color.js b/lib/color.js
index cf252d2..a49675d 100644
--- a/lib/color.js
+++ b/lib/color.js
@@ -263,17 +263,10 @@ class ColorXY {
             blue = 1.0;
         }

-        if (isNaN(red)) {
-            red = 0;
-        }
-
-        if (isNaN(green)) {
-            green = 0;
-        }
-
-        if (isNaN(blue)) {
-            blue = 0;
-        }
+        // This fixes situation when due to computational errors value get slightly below 0, or NaN in case of zero-division.
+        red = (isNaN(red) || red < 0) ? 0 : red;
+        green = (isNaN(green) || green < 0) ? 0 : green;
+        blue = (isNaN(blue) || blue < 0) ? 0 : blue;

         return new ColorRGB(red, green, blue);
     }
```

To fix the tests you should use `.rounded()` method, like I used in test for `ColorRGB.toXY()`:

```diff
diff --git a/test/color.test.js b/test/color.test.js
index 36d4339..c9e5663 100644
--- a/test/color.test.js
+++ b/test/color.test.js
@@ -102,11 +102,11 @@ describe('lib/color.js', () => {
         });

         test.each([
-            ['red', {x: 0.7006, y: 0.299}, {red: 1.0, green: 0, blue: 0}],
+            ['red', {x: 0.7006, y: 0.2993}, {red: 1.0, green: 0, blue: 0}],
             ['green', {x: 0.1724, y: 0.7468}, {red: 0, green: 1.0, blue: 0}],
             ['blue', {x: 0.1355, y: 0.0399}, {red: 0, green: 0, blue: 1.0}],
         ])('.toRGB() - %s', (_name, xy, rgb) => {
-            expect(libColor.ColorXY.fromObject(xy).toRGB().toObject()).toStrictEqual(rgb);
+            expect(libColor.ColorXY.fromObject(xy).toRGB().rounded(4).toObject()).toStrictEqual(rgb, 4);
         });

         test.each([
````